packages/plugin-compiler/src/test/resources/modules/expected/00_ValidateIrBeforeLowering.ir

```diff
@@ -9,12 +9,12 @@ MODULE_FRAGMENT name:<main>
         BLOCK_BODY
           DELEGATING_CONSTRUCTOR_CALL 'public constructor <init> () [primary] declared in kotlin.Any'
           INSTANCE_INITIALIZER_CALL classDescriptor='CLASS CLASS name:A modality:OPEN visibility:public superTypes:[io.realm.runtimeapi.RealmModel; io.realm.runtimeapi.RealmModelInternal]'
-      CLASS OBJECT name:Companion modality:FINAL visibility:public [companion] superTypes:[kotlin.Any; io.realm.internal.RealmObjectCompanion]
+      CLASS OBJECT name:Companion modality:OPEN visibility:public [companion] superTypes:[kotlin.Any; io.realm.internal.RealmObjectCompanion]
         $this: VALUE_PARAMETER INSTANCE_RECEIVER name:<this> type:modules.input.A.Companion
         CONSTRUCTOR visibility:private <> () returnType:modules.input.A.Companion [primary]
```

Comment:
why it's `OPEN`?

Reply:
I think it is just open when on of the fields are. I just grabbed the local for adding a property from some existing code. I will _final_ize it. I guess we should probably protect as must of the generate API.